### ISSUE TITLE
Implicit @* in deprecate misses prerelease packages

### ISSUE THREAD

**Issue by isaacs** [npm/npm#18282]
#### What's going wrong?

Via twitter: https://twitter.com/jsumners79/status/900446451825823752

Answer:  https://twitter.com/izs/status/900476230532685824

#### How can the CLI team reproduce the problem?

Publish a package with a prerelease version.
Try to deprecate that package with `npm deprecate packagename "message"`
Package is not deprecated.  (Nothing happens.)

### supporting information:

 - `npm -v` prints: 5.4.0

Code here: https://github.com/npm/npm/blob/latest/lib/deprecate.js#L40-L42

Probably there should be a way to tell `npm.registry.deprecate` "deprecate all of the things, even the things that `*` semver range won't match".  Or maybe it should just interpret `*` differently in this instance to mean "all of the things" rather than "all the things that match the range".

**Comment by jsumners** [comment]
I agree. The docs are mostly clear that leaving off any specific version, or version qualifier, should target all versions of the package. This was bewildering behavior.
